tests/test_pdb_chainbows.py

```diff
@@ -0,0 +1,140 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Copyright 2020 João Pedro Rodrigues
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Unit Tests for `pdb_chainbows`.
+"""
+
+import os
+import sys
+import unittest
+
+from config import data_dir
+from utils import OutputCapture
+
+
+class TestTool(unittest.TestCase):
+    """
+    Generic class for testing tools.
+    """
+
+    def setUp(self):
+        # Dynamically import the module
+        name = 'pdbtools.pdb_chainbows'
+        self.module = __import__(name, fromlist=[''])
+
+    def exec_module(self):
+        """
+        Execs module.
+        """
+
+        with OutputCapture() as output:
+            try:
+                self.module.main()
+            except SystemExit as e:
+                self.retcode = e.code
+
+        self.stdout = output.stdout
+        self.stderr = output.stderr
+
+        return
+
+    def test_default(self):
+        """$ pdb_chainbows data/dummy.pdb"""
+
+        fpath = os.path.join(data_dir, 'dummy.pdb')
+        sys.argv = ['', fpath]
+
+        # Execute the script
+        self.exec_module()
+
+        # Validate results
+        self.assertEqual(self.retcode, 0)  # ensure the program exited OK.
+        # CONECTs are ignored by issue #72, expected only 204 lines
+        self.assertEqual(len(self.stdout), 204)
+
+        # Check chains were reassigned properly
+        expected = ['A'] * 51 + ['B'] * 56 + ['C'] * 49 + ['D'] * 20
+        chains = [
+            l[21] for l in self.stdout if l.startswith('ATOM')
+        ]
+        self.assertEqual(chains, expected)
+
+        # Check TERs
+        expected = ['A', 'B', 'C']
+        chains = [
+            l[21] for l in self.stdout if l.startswith('TER')
+        ]
+        self.assertEqual(chains, expected)
+
+        # Check HETATM were ignored and warning was issued
+        self.assertEqual(len(self.stderr), 1)  # one warning message
+        self.assertEqual(self.stderr[0][:40], 'WARNING!! HETATM will not be reassigned.')
```

Comment:
If you implement the dictionary strategy as above, this part of the test needs to be updated.

Reply:
Done.